### DISCUSSION TITLE
I am building the OPS framework in my computer

### DISCUSSION THREAD

**Discussion by lcmj2803** [omnetpp/omnetpp#1215] [Q&A] [answered]
The link is below
[https://github.com/ComNets-Bremen/OPS](url)
(https://github.com/user-attachments/assets/103a5c90-9b64-4552-9e26-53e1f8eed603)

I am using everything in the repo. I am running it on a Ubuntu 22.04 machine. I downloaded every prerequisite but in the last step where I have to build OPS, I get the error on the 4 screenshots below.

![Screenshot from 2024-09-17 03-26-37](https://github.com/user-attachments/assets/0a66dc99-c6ee-48fd-84dd-0b962f20a1d9)
![Screenshot from 2024-09-17 03-27-28](https://github.com/user-attachments/assets/08a6f746-e29b-4878-9caa-80972f481031)
![Screenshot from 2024-09-17 03-27-51](https://github.com/user-attachments/assets/fef2682b-1ac1-44ec-97f0-61b86274778b)
![Screenshot from 2024-09-17 03-28-32](https://github.com/user-attachments/assets/942c53ed-25be-4999-b467-306497f09ab0)

I have been trying to download osg-earth to solve it, but I get the error shown in the image below

![osg_error](https://github.com/user-attachments/assets/1a72268d-7703-432b-9ca9-398c0501d312)

**Comment by avarga** [comment]
Not sure why OPS needs OSG, I'm fairly sure it doesn't contain any 3D
visualization.

Have you tried installing opp_env? Should take just one command (after you
installed opp_env itself):

$ opp_env install ops_allinone-latest

**Reply by lcmj2803** [reply]
Hey @avarga  thanks for your quick response !!
I have to do that in the terminal and then I run OPS once again?

**Comment by avarga** [comment, marked as answer]
opp_env installs everything, including OMNeT++ and its dependencies, so you
may delete your existing OPS and OMNeT++ directories. You won't need them.

Install opp_env:

$ pip3 install --upgrade pip
$ pip3 install opp_env

opp_env requires Nix, so install nix from nixos.org too.

Installing opp_env might look like a hassle, but it allows you one-command
unattended installation of almost any version of almost any OMNeT++ model.
See more here: https://github.com/omnetpp/opp_env

Then you can run the following command to download and install OPS with a
matching OMNeT++ version:

$ mkdir ops-all && cd ops-all && opp_env init && opp_env install
ops_allinone-latest

This will download and compile everything, so expect it to take a while.
You can use the resulting installation from the opp_env shell, so enter
this:

$ opp_env shell

Then, if you cd into OPS's simulations folder, you can run simulations like
this:

$ cd simulations
$ ../ops-simu omnetpp-ops.ini -n '../src:.:../modules/inet/src'

Hope this helps,
Andras

**Reply by lcmj2803** [reply]
It worked like a charm thanks!!!! Now I am going to start creating my own OPS  networks
